docs/angular.md

```diff
@@ -0,0 +1,227 @@
+    npm i --save zod
+    ```
+
+1.  If you didn't install the Genkit CLI globally, you can install it as a
+    development dependency. The tsx tool is also recommended, as it makes
+    testing your code more convenient. Both of these dependencies are optional,
+    however.
+
+    ```posix-terminal
+    npm i --save-dev genkit-cli tsx
+    ```
+
+## Define Genkit flows
+
+Create a new file in your Angular project to contain your Genkit flows: for
+example, `src/genkit.ts`. This file can contain your flows without
+modification.
+
+For example:
+
+```ts
+import { gemini20Flash, googleAI } from "@genkit-ai/googleai";
+import { genkit } from "genkit";
+import { z } from "zod";
+
+const ai = genkit({
+  plugins: [googleAI()],
+  model: gemini20Flash,
+});
+
+export const menuSuggestionFlow = ai.defineFlow(
+  {
+    name: "menuSuggestionFlow",
+    inputSchema: z.string(),
+    outputSchema: z.string(),
+  },
+  async (restaurantTheme) => {
+    const { text } = await ai.generate(`Invent a menu item for a ${restaurantTheme} themed restaurant.`);
+    return text;
+  }
+);
+```
+
+## Add server routes
+
+Add the following imports to `src/server.ts`:
+
+```ts
+import { expressHandler } from '@genkit-ai/express';
+import { menuSuggestionFlow } from './genkit';
+```
+
+Add the following line following your `app` variable initialization:
+
+```ts
+app.use(express.json());
+```
+
+Then, add a route to serve your flow:
+
+```ts
+app.post('/menu', expressHandler(menuSuggestionFlow));
+```
+
+## Call your flows
+
+Now, you can run your flows from your client application.
+
+For example, you can replace the contents of
+`src/app/app.component.ts` with the following:
+
+```ts
+import { Component, resource, signal } from '@angular/core';
+import { FormsModule } from '@angular/forms';
+import { runFlow } from 'genkit/beta/client';
+
+@Component({
+  selector: 'app-root',
+  imports: [FormsModule],
+  templateUrl: './app.component.html',
+})
+export class AppComponent {
+  menuInput = '';
+  theme = signal('');
+
+  menuResource = resource({
+    request: () => this.theme(),
+    loader: ({request}) => runFlow({ url: 'menu', input: request })
+  });
+}
+```
+
+Make corresponding updates to `src/app/app.component.html`:
+
+```ts
+<h3>Generate a custom menu item</h3>
+<input type="text" [(ngModel)]="menuInput" />
+<button (click)="this.theme.set(menuInput)">Generate</button>
+<br />
+@if (menuResource.isLoading()) {
+  Loading...
+} @else {
+  <pre>{{menuResource.value()}}</pre>
+}
+```
+
+## Test your app locally
+
+If you want to run your app locally, you need to make credentials for the model
+API service you chose available.
+
+- {Gemini (Google AI)}
+
+  1.  Make sure Google AI is
+      [available in your region](https://ai.google.dev/available_regions).
+
+  1.  [Generate an API key](https://aistudio.google.com/app/apikey) for the
+      Gemini API using Google AI Studio.
+
+  1.  Set the `GOOGLE_GENAI_API_KEY` environment variable to your key:
```

Comment:
what's the difference between this and `GEMINI_API_KEY`

Reply:
Updated to `GEMINI_API_KEY`